medcat/utils/checkpoint.py

```diff
@@ -60,65 +54,118 @@ def count(self) -> int:
         return self._count
 
     @property
-    def metadata(self) -> Optional[Dict]:
-        return self._metadata
+    def dir_path(self) -> str:
+        return self._dir_path
 
     @classmethod
-    def restore(cls, dir_path: str) -> "Checkpoint":
+    def from_last_training(cls, parent_dir_path: str) -> "Checkpoint":
+        if not os.path.isdir(parent_dir_path):
+            raise ValueError(f"Checkpoint folder passed in does not exist: {parent_dir_path}")
+        ckpt_parent_folders = os.listdir(parent_dir_path)
+        if not ckpt_parent_folders:
+            raise ValueError("No existing training found")
+        ckpt_parent_folders.sort()
+        ckpt_file_path = os.path.abspath(os.path.join(parent_dir_path, ckpt_parent_folders[-1]))
+        checkpoint = cls.load(ckpt_file_path)
+        return checkpoint
+
+    @classmethod
+    def load(cls, dir_path: str) -> "Checkpoint":
```

Comment:
I don't see the use case for restore / load and populate being separated. A checkpoint will always be restored and populated so why break up the functionality?

Reply:
Individual methods removed.